File: rotateArray.cpp
#include <iostream>
using namespace std;


 void rotate(int arr[], int d , int size) {
        int temp[size];
        for(int i=0;i<size;i++)
            temp[i]=arr[i];
        for(int i=0;i<size;i++)
        {
            if(i-d>=0)
            arr[i-d]=temp[i];
            else
            arr[size+(i-d)]=temp[i];
        }
    }

int main() {
	int testCases;
	cin>>testCases;
		while(testCases-->0){
		    //taking size of array and creating array
		    int size,rotateIndex;
		    cin>>size>>rotateIndex;
		    int arr[size];
		    //taking input of arrays
		    for(int i=0;i<size;i++){
		        cin>>arr[i];
		    }
		    rotate(arr,rotateIndex,size);
		    for(int i=0;i<size;i++){
		       cout<<arr[i]<<" ";
		    }
		    cout<<"\n"; // don't use endl it is slower
		}
	return 0;
}

File: rotateArray.java
/*package whatever //do not write package name here */

import java.util.*;
import java.lang.*;
import java.io.*;

class GFG {
    
    
     public static void rotate(int[] arr, int d) {
        int[] temp= new int[arr.length];
        for(int i=0;i<arr.length;i++)
            temp[i]=arr[i];
        for(int i=0;i<arr.length;i++)
        {
            if(i-d>=0)
            arr[i-d]=temp[i];
            else
            arr[arr.length+(i-d)]=temp[i];
        }
    }
	public static void main (String[] args) {
	    Scanner sc = new Scanner(System.in);
		int testCases = sc.nextInt();
		while(testCases-->0){
		    //taking size of array and creating array
		    int size = sc.nextInt();
		    int rotateIndex = sc.nextInt();
		    int arr[] = new int[size];
		    //taking input of arrays
		    for(int i=0;i<size;i++){
		        arr[i] = sc.nextInt();
		    }
		    rotate(arr,rotateIndex);
		    for(int i=0;i<size;i++){
		        System.out.print(arr[i]+" ");
		    }
		    System.out.println();
		}
	}
}
